schema/schema_validator.py

```diff
@@ -66,65 +67,62 @@ def validate_json_file(self, schema_and_data_paths):
         except FileNotFoundError as err:
-            logging.debug('  Cannot open data file %s.\n   Err = %s', data_file_path, err)
-
+            logging.fatal('  Cannot open data file %s.\n   Err = %s', data_file_path, err)
             result_data['result'] = False
             result_data['error'] = err
-            return result_data
+            exit(1)
 
-      # Get the schema file and validate the data against it
+        # Get the schema file and validate the data against it
         try:
             schema = json.load(schema_file)
         except json.decoder.JSONDecodeError as err:
-            logging.error('Bad JSON schema: %s', schema_file_path)
-            logging.error('  Error is %s', err)
             result_data['result'] = False
             result_data['error'] = err
-            return result_data
+            logging.error('Bad JSON schema: %s', schema_file_path)
+            logging.fatal('  Error is %s', err)
+            exit(1)
 
         try:
             data_to_check = json.load(data_file)
         except json.decoder.JSONDecodeError as err:
-            logging.error('Bad JSON data: %s', ddata_file_path)
-            logging.error('  Error is %s', err)
+            # Cannot get the file
             result_data['result'] = False
             result_data['error'] = err
-            return result_data
+            logging.error('Bad JSON data: %s', data_file_path)
+            logging.fatal('  Error is %s', err)
+            exit(1)
+
         # Now check this!
         try:
             validate(data_to_check, schema)
             # Everything worked!
             result_data['result'] = True
-            result_data['error'] = None
-            return result_data
         except ValidationError as err:
-            logging.error('ValidationError for test output %s and schema %s',
-                          data_file_path, schema_file_path )
-            logging.error('  Error = %s', err)
             result_data['result'] = False
             result_data['error'] = err
-            return result_data
+            logging.error('ValidationError for test output %s and schema %s',
+                          data_file_path, schema_file_path)
+            logging.fatal('  Error = %s', err)
+            exit(1)
         except exceptions.SchemaError as err:
+            result_data['result'] = False
+            result_data['error'] = err
             logging.error('SchemaError: Cannot validate with test output %s and schema %s. ',
                           data_file_path, schema_file_path)
             logging.error('  Error = %s', err)
-            result_data['result'] = False
-            result_data['error'] = err
-            return result_data
-        except BaseException as err:
-            logging.error('Another failure: %s', err)
-            result_data['result'] = False
-            result_data['error'] = err
-            return result_data
+            logging.fatal('Another failure: %s', err)
```

Comment:
remove the line above that says `logging.error` since it's redundant with this line, and it should be fatal anyways since you're exiting early.

Reply:
Done